spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -376,6 +363,12 @@ public Scan build() {
   }
 
   private Scan buildBatchScan() {
+    Schema expectedSchema = schemaWithMetadataColumns();
+    return new SparkBatchQueryScan(
+        spark, table, buildIcebergBatchScan(false), readConf, expectedSchema, filterExpressions);
+  }
+
+  private org.apache.iceberg.Scan buildIcebergBatchScan(boolean withStats) {
```

Comment:
nit: Do we need the fully qualified type?

Reply:
Yea, we need the fully qualified type because we have already `import org.apache.spark.sql.connector.read.Scan`